bin/controller.ts

```diff
@@ -2,37 +2,36 @@
 
 import bonjour from 'bonjour'
 
-import { once } from 'events'
-
 import {
 	supplyClient,
 	Client,
-	Http2Client,
+	createSSHConnection,
+	GenericClient,
 	NormalJob,
 	HeapJobOrderer,
 	Job,
 } from '../src'
 
 const zeroconf = bonjour()
 
-function clientDone() {
-	console.log('Finished')
-
-	// The mDNS socket apparently has no way to tell that it's not needed.
-	zeroconf.destroy()
-}
-
 const job3: Job = new NormalJob('third')
 const job4: Job = new NormalJob('fourth')
 const job2: Job = new NormalJob('second', new Set([job3]))
 const job1: Job = new NormalJob('first', new Set([job4, job2]))
 const job5: Job = new NormalJob('fifth', new Set([job1]))
 
-const client: Client = new Http2Client(new HeapJobOrderer([job5]))
-client.on('progress', (finishedJob) => {
-	console.log(`${finishedJob.getName()} completed\n`)
+const client: Client = new GenericClient(
+	createSSHConnection,
+	process,
+	new HeapJobOrderer([job5]),
+)
+client.on('error', console.error)
+client.on('progress', (job, result) => console.log(job, result.status))
```

Comment:
It seems that job's `toString()` method isn't defined. 
If I understand JS string representations correctly, we'll only print out something like `[NormalJob]`, which doesn't really give any information to the user. Is this a problem, or am I wrong?

Reply:
`console.log` is special. It prints out the whole object nicely with colors. Though, in this case, `NormalJob` does have a `toString` so it'll print that.